config/groovy/common.groovy

```diff
@@ -182,12 +182,47 @@ class common {
         addRemote(itemName, defaultRemote, "https://github.com/${githubDefaultHome}/${itemName}.git")
     }
 
+    /**
+     * Check if an item was updated within the provided time limit
+     * @param fetchHead the item's FETCH_HEAD file in the .git directory
+     * @param timeLimit the time limit for considering something recently updated, for example: use(groovy.time.TimeCategory){ 10.minute }
+     */
+    def isRecentlyUpdated(File fetchHead, def timeLimit){
+        Date lastUpdate = new Date(fetchHead.lastModified())
+        def recentlyUpdated = use(groovy.time.TimeCategory){
+            def timeElapsedSinceUpdate = new Date() - lastUpdate
+            if (timeElapsedSinceUpdate < timeLimit){
+                return true
+            } else {
+                return false
+            }
+        }
+    }
+
     /**
      * Update a given item.
      * @param itemName the name of the item to update
      */
-    def updateItem(String itemName) {
+    def updateItem(String itemName, boolean skipRecentUpdates = false) {
         File targetDir = new File(targetDirectory, itemName)
+        def searchString = itemName
+        if (!Character.isLetterOrDigit(itemName.charAt(0))){   
+            println color ("Skipping update for $itemName: starts with non-alphanumeric symbol", Ansi.YELLOW)
+            return
+        }
+        def inGitIgnore = false
+        new File(".gitignore").eachLine{ line -> 
+            // match if line is exactly same os itemName or has trailing '/' but
+            // not if has further subdirectories
+            if ((line ==~ searchString) || (line ==~ "$searchString/")){
+                inGitIgnore = true
+                return
+            }
+        }
+        if (inGitIgnore){
```

Comment:
Please turn this into a method and use it inline in the check:

```suggestion
        if (isGitIgnored(itemName)) {
```

Reply:
Oof, I mean to revert this as well. This isn't actually going to be used any more, I took out the gitignore processing in favor of just ignoring directories starting with non-alpha-numeric characters since I can't truly honor the gitignore